compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParserErrorHandler.java

```diff
@@ -604,14 +604,14 @@
             { ParserRuleContext.COMMA, ParserRuleContext.CLOSE_BRACE };
 
     private static final ParserRuleContext[] ERROR_MATCH_PATTERN_OR_CONST_PATTERN =
-            { ParserRuleContext.OPEN_PARENTHESIS, ParserRuleContext.MATCH_PATTERN_END };
+            { ParserRuleContext.OPEN_PARENTHESIS, ParserRuleContext.MATCH_PATTERN_RHS };
 
     private static final ParserRuleContext[] ERROR_MATCH_PATTERN_ERROR_KEYWORD_RHS =
             { ParserRuleContext.OPEN_PARENTHESIS, ParserRuleContext.TYPE_REFERENCE };
 
     private static final ParserRuleContext[] ERROR_ARG_LIST_MATCH_PATTERN_START =
             { ParserRuleContext.CONSTANT_EXPRESSION, ParserRuleContext.VAR_KEYWORD,
-                    ParserRuleContext.ERROR_FIELD_MATCH_PATTERN, ParserRuleContext.CLOSE_PARENTHESIS };
+                    ParserRuleContext.ERROR_FIELD_MATCH_PATTERN };
```

Comment:
Why we do need to exclude CLOSE_PARENTHESIS here?
According to spec https://ballerina.io/ballerina-spec/spec.html#error-arg-list-match-pattern `error-arg-list-match-pattern` is optional?

Reply:
Fixed in  81ed241